### PR TITLE
[ITG-24] Create, view and delete subtasks.

### PR DESCRIPTION
**Pull request by gregori0o**
I have one small problem with layout. In details view first subtask card is higher than other subtasks.

Edit: Solved.

### CODE REVIEW

File: cabra/src/pages/components/MiniTaskDelete.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import "twin.macro";
+
+import NavigationButton from "./NavigationButton";
+import { TrashIcon } from "@heroicons/react/solid";
+
+interface Props {
+  title: string;
+  relationId: string;
+  taskId: string;
+  onClickDelete: CallableFunction;
```
Comment (john-sonz) [inline cabra/src/pages/components/MiniTaskDelete.tsx:10]: You can be more specific with typing here
```suggestion
  onClickDelete: (relationId: string, taskId:string) => void ;
```

File: cabra/src/pages/components/SubtasksListEdit.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    const inputs = {
+      ...emptyTask,
+      title: val,
+    };
+    addSubtask.mutateAsync(inputs);
+  };
+
+  const removeRelation = useMutation((relationId: string) =>
+    deleteRelation(relationId)
+  );
```
Comment (john-sonz) [inline cabra/src/pages/components/SubtasksListEdit.tsx:65]:
```suggestion
  const removeRelation = useMutation(deleteRelation);
```

**Review by john-sonz — COMMENTED** [review]
Good job overall! I left suggestions for some small improvements but nothing major.

File: cabra/src/pages/components/SubtasksListEdit.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      client.invalidateQueries(getTasks.cacheKey);
+      client.invalidateQueries(getTask.cacheKey(parentId));
+      client.invalidateQueries(getSubtasks.cacheKey(parentId));
+    },
+  });
+
+  const removeSubtask = (relationId: string, subtaskId: string) => {
+    removeRelation.mutateAsync(relationId);
+    removeTask.mutateAsync(subtaskId);
+  };
```
[the remark this answers is not in the document]
Comment (gregori0o) [reply cabra/src/pages/components/SubtasksListEdit.tsx:77]: It isn't working like that. I can't delete subtasks if relation is in database. But I think is good idea to delete all records in `TaskRelations` table related with task when we remove this task. I will implement this.

[the remark this answers is not in the document]
Comment (gregori0o) [reply cabra/src/pages/components/SubtasksListEdit.tsx]: Now, if we delete task, all relations will be deleted  and all subtasks will be deleted.

**Review by john-sonz — APPROVED** [review]
Awesome! 🚀

**Pull request merged**